test/object-store/sync/client_reset.cpp

```diff
@@ -3091,7 +3091,7 @@ TEMPLATE_TEST_CASE("client reset collections of links", "[sync][pbs][client rese
                              {Remove{dest_pk_1}, Remove{dest_pk_2}}, {dest_pk_3, dest_pk_4});
         }
     }
-    else if constexpr (test_type_is_set) {
+    if constexpr (test_type_is_set) {
```

Comment:
With the `else` here, GCC determined that this `if` was unreachable without ever evaluating `test_type_is_set` and so produced an unused variable warning. This is IMO a bug in GCC, but removing the else doesn't change anything.